controllers/articles.js

```diff
@@ -0,0 +1,195 @@
+import crypto from 'crypto';
+import models from '../models/index';
+
+const tempUser = '2dd9e22b-d19d-4501-a6b5-00bb4ebd9b3e'; // This line has to be deleted.
+const { article, User } = models;
+
+const articles = {
+
+  /*
+   * Deleting a post based to its slug.
+   *
+   * For directly publishing an article, status flag has to be passed
+   * into the request body object otherwise the article will be in drafted.
+   */
+  createAnArticle: async (req, res) => {
+    try {
+      const {
+        title, body, author, status, tagList,
+      } = req.body;
+      const flag = status === undefined ? 'Draft' : 'Published';
+      const tags = req.is('application/json') ? tagList : JSON.parse(tagList);
+
+      const slug = `${title.toLowerCase().split(' ').join('-').substring(0, 20)}${crypto.randomBytes(5).toString('hex')}`;
+      const description = body.substring(0, 100);
+
+      const queryArticle = await article.create({
+        title, body, author, slug, description, status: flag, tagList: tags,
+      });
+      return res.status(201).send({
+        status: res.statusCode,
+        article: {
+          id: queryArticle.id,
+          title: queryArticle.title,
+          description: queryArticle.description,
+          body: queryArticle.body,
+          slug: queryArticle.slug,
+          tags: [queryArticle.taglines]
+        }
+      });
+    } catch (err) {
+      throw err;
+    }
+  },
+
+  // update an Article
+  updateAnArticle: async (req, res) => {
+    try {
+      const oldSlug = req.params.slug;
+      const { title, body } = req.body;
```

Comment:
Can't someone update an article's tags?

Reply:
@avpaul please be more descriptive on this...